tests/integration/utils/db_faker.py

```diff
@@ -0,0 +1,257 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Utility for faking a cylc run database.
+
+This allows us to put suites into strange states without having to
+make the series of events leading to that state actually happen.
+
+This allows us to test a whole range of exotic and hard or even impossible to
+reproduce states.
+
+These utilities are not intended for direct use by tests.
+Use the fixtures provided in the conftest instead.
+
+"""
+
+from collections import namedtuple
+import getpass
+
+from cylc.flow.rundb import CylcSuiteDAO
+from cylc.flow.task_state import TASK_STATUSES_ORDERED
+
+__all__ = ('fake_db')
+
+
+UNSET = '*unset*'
+
+TASK_FIELDS = {
+    # these fields you must define
+    'name': UNSET,
+    'cycle': UNSET,
+    'status': UNSET,
+    # these fields we can guess for you
+    'jobs': None,
+    'submit_num': None,
+    # these fields you need to provide yourself
+    # (else you get the defaults)
+    'spawned': False,
+    'is_held': False,
+    'time_created': None,
+    'time_updated': None,
+}
```

Comment:
It's set up so that you don't have to provide full information about a task.

For example if you provide `Task('foo', '1', 'running')` then we can infer:

* `submit_num=1`
* `jobs=Job('running', name='foo', cycle='1', submit_num='1', try_num='1')`